spec/example_spec.rb

```diff
@@ -1,6 +1,30 @@
 require_relative 'spec_helper'
 
 describe 'Asciidoctor::PDF::Converter - Example' do
+  it 'should keep block together when block has the unbreakable option' do
+    to_file = to_pdf_file <<~EOS, 'example-unbreakable-option.pdf'
+    Make it rain.footnote:[money]
```

Comment:
Even without `push_scratch doc if scratch?` and `pop_scratch doc if scratch?` the footnotes are not duplicated so I'm not sure how to test this case.

Reply:
The footnotes have to be inside the open block.

Reply:
I did add a footnote before the open block, inside the open block and after the open block :thinking:

Reply:
Aha. The scenario is that the footnote is in the open block, but not inside another nested block. In the test, the footnote is in an admonition block, which masks the problem.